code/game/machinery/computer/camera_console.dm

```diff
@@ -219,32 +222,55 @@
 	. = ..()
 	if(action != "switch_camera")
 		return
-	broadcastingcamera = null
-	if (!istype(current, /obj/structure/machinery/camera/correspondent))
+	if(broadcastingcamera)
+		clear_camera()
+	if(!istype(current, /obj/structure/machinery/camera/correspondent))
 		return
 	var/obj/structure/machinery/camera/correspondent/corr_cam = current
-	if (!corr_cam.linked_broadcasting)
+	if(!corr_cam.linked_broadcasting)
 		return
 	broadcastingcamera = corr_cam.linked_broadcasting
 	RegisterSignal(broadcastingcamera, COMSIG_BROADCAST_GO_LIVE, PROC_REF(go_back_live))
 	RegisterSignal(broadcastingcamera, COMSIG_PARENT_QDELETING, PROC_REF(clear_camera))
+	RegisterSignal(broadcastingcamera, COMSIG_BROADCAST_HEAR_TALK, PROC_REF(transfer_talk))
+	RegisterSignal(broadcastingcamera, COMSIG_BROADCAST_SEE_EMOTE, PROC_REF(transfer_emote))
 
 /obj/structure/machinery/computer/cameras/wooden_tv/broadcast/ui_close(mob/user)
 	. = ..()
-	if (!current && broadcastingcamera)
+	if(!broadcastingcamera)
+		return
+	if(!current)
 		clear_camera()
 
 /obj/structure/machinery/computer/cameras/wooden_tv/broadcast/proc/clear_camera()
 	SIGNAL_HANDLER
-	UnregisterSignal(broadcastingcamera, list(COMSIG_BROADCAST_GO_LIVE, COMSIG_PARENT_QDELETING))
+	UnregisterSignal(broadcastingcamera, list(COMSIG_BROADCAST_GO_LIVE, COMSIG_PARENT_QDELETING, COMSIG_BROADCAST_HEAR_TALK, COMSIG_BROADCAST_SEE_EMOTE))
 	broadcastingcamera = null
 
 /obj/structure/machinery/computer/cameras/wooden_tv/broadcast/proc/go_back_live(obj/item/device/camera/broadcasting/broadcastingcamera)
 	SIGNAL_HANDLER
-	if (current.c_tag == broadcastingcamera.get_broadcast_name())
+	if(current.c_tag == broadcastingcamera.get_broadcast_name())
 		current = broadcastingcamera.linked_cam
 		SEND_SIGNAL(src, COMSIG_CAMERA_SET_TARGET, broadcastingcamera.linked_cam, broadcastingcamera.linked_cam.view_range, broadcastingcamera.linked_cam.view_range)
 
+/obj/structure/machinery/computer/cameras/wooden_tv/broadcast/proc/transfer_talk(obj/item/camera, mob/living/sourcemob, message, verb = "says", datum/language/language, italics = FALSE, show_message_above_tv = FALSE)
+	SIGNAL_HANDLER
+	if(show_message_above_tv)
+		langchat_speech(message, get_mobs_in_view(7, src), language, sourcemob.langchat_color, FALSE, LANGCHAT_FAST_POP, list(sourcemob.langchat_styles))
+	for(var/datum/weakref/user_ref in concurrent_users)
+		var/mob/user = user_ref?.resolve()
```

Comment:
```suggestion
	for(var/datum/weakref/user_ref in concurrent_users)
		var/mob/user = user_ref.resolve()
```
Since you aren't using `as anything` the `?` usage is unnecessary.